main.tf

```diff
@@ -52,7 +52,7 @@ resource "aws_globalaccelerator_listener" "this" {
 ################################################################################
 
 resource "aws_globalaccelerator_endpoint_group" "this" {
-  for_each = { for k, v in var.listeners : k => v if var.create && var.create_listeners && can(var.listeners[k].endpoint_group) }
+  for_each = { for k, v in var.listeners : k => v if var.create && var.create_listeners && length(lookup(var.listeners[k], "endpoint_group", {})) >= 0 }
```

Comment:
`can`, `try`, (and `jsonencode()` for this matter) do late-resolving of the value, which led to the famous error `Invalid for_each argument - The "for_each" value depends on resource attributes that cannot be determined until apply`.

Using `lookup()` is old-fashion, but it works because it resolves early to a known value. This is exactly what we need if we need to use in `for_each`.